orc8r/tools/fab/vagrant.py

```diff
@@ -12,61 +12,59 @@
 """
 import os
 import os.path
+from typing import Dict, Tuple
 
-from fabric.api import cd, env, local
+from fabric import Connection
 
 
-def __ensure_in_vagrant_dir():
+def _ensure_in_vagrant_dir(c: Connection) -> None:
     """
     Error out if there is not Vagrant instance associated with this directory
     """
-    pwd = local('pwd', capture=True)
+    pwd = c.run('pwd', hide=True).stdout.strip()
     if not os.path.isfile(pwd + '/Vagrantfile'):
         # check if we are on a vagrant subdirectory
-        with cd(pwd):
-            if not local('vagrant validate', capture=True):
-                print("Error: Vagrantfile not found")
-                exit(1)
+        if not c.run('vagrant validate', hide=True, warn=True).return_code == 0:
+            print("Error: Vagrantfile not found")
+            exit(1)
```

Comment:
I found the use of `exit(...)`, and `sys.exit(...)`, and `raise` a bit unclear throughout the fab files. Maybe it is worth making the use more consistent. At this point they only appear to convenient ways to exit the fabric scripts. We could also think more about the exit codes. In the if statement above, we explicitly check against a return code. But why do we exit with 1 instead just using that return code?

Reply:
I added the general overhaul of `exit()`/`sys.exit` to the refactoring ticket #14909. Regarding this occurrence: I checked the return codes of `vagrant validate`, which gives you `1` if there are syntax errors or no vagrant file is present. To give a more broad message, I adapted the print output and exit the explicit return code as you suggested.